### PR TITLE
Sliding drawer transition

### PR DESCRIPTION
**Pull request by jamiesmyln**
This PR includes: 

- the Drawer component 
- Adding Transition to Modal for the animation 
-  Datepicker modal styling fix.

### CODE REVIEW

File: packages/react-magma-dom/src/components/Drawer/Drawer.test.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { axe } from 'jest-axe';
+import { Drawer } from '.';
+import { render } from '@testing-library/react';
+
+const TEXT = 'Test Text';
+
+describe('Drawer', () => {
```
Comment (joasmi723) [inline packages/react-magma-dom/src/components/Drawer/Drawer.test.js:8]: These just look like the default tests, there should probably be more.

File: packages/react-magma-dom/src/components/Drawer/useDrawer.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/**
```
Comment (joasmi723) [inline packages/react-magma-dom/src/components/Drawer/useDrawer.ts:1]: If we're not using this file, we probably should just delete it.

File: packages/react-magma-dom/src/components/Drawer/Drawer.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * @children required
+ */
+export interface DrawerProps extends ModalProps {
+  testId?: string;
+  /**
+   * Style properties for the drawer
+   */
+  style?: React.CSSProperties;
```
Comment (joasmi723) [inline packages/react-magma-dom/src/components/Drawer/Drawer.tsx:30]: I wonder if we should call this property `drawerStyle` to distinguish it from `containerStyle`.  I believe that's what we do in other components that take multiple style props.

File: packages/react-magma-dom/src/components/Modal/Modal.tsx [hunk 4]
```diff
@@ -372,89 +342,87 @@ export const Modal = React.forwardRef<HTMLDivElement, ModalProps>(
...
-                isInverse={false}
-                ref={ref}
-                theme={theme}
-              >
+    return (
+      <>
+        <Global
+          styles={css`
+            html {
+              overflow: hidden;
```
Comment (joasmi723) [inline packages/react-magma-dom/src/components/Modal/Modal.tsx:350]: It looks like this is getting applied even when the modal or drawer isn't open.  As a result, I can't scroll on the modal and drawer docs pages.

File: website/react-magma-docs/src/pages/api/drawer.mdx [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+A container with a transition
+
+## Basic Usage
+
+```typescript
+import React from 'react';
+import { Drawer } from 'react-magma-dom';
+
+export function Example() {
+  return <Drawer>Drawer content</Drawer>;
```
Comment (joasmi723) [inline website/react-magma-docs/src/pages/api/drawer.mdx:19]: These examples don't have a way to trigger the drawer.  Can you add a button like for the storybook stories?

Also, we should probably show an example where we set the drawer position.

**Review by joasmi723 — CHANGES_REQUESTED** [review]

File: packages/react-magma-dom/src/components/Drawer/Drawer.test.js (hunk 1 above)
Comment (jamiesmyln) [reply packages/react-magma-dom/src/components/Drawer/Drawer.test.js:10]: Added tests to cover the 4 positions

**Comment by aaronmaturen** [comment]
⚠️ No Changeset found 😱

**Pull request merged**